### ISSUE TITLE
Player cannot accept quest, asks to contact admin.

### ISSUE THREAD

**Issue by lyimpike** [PikaMug/Quests#946]
**What is your Quests version / build number (do _not_ say "latest")?**
3.8.0

**Describe the problem and how to replicate it. Any console errors to post _using Pastebin.com_?**
When a fresh player tries to join the server and accept a quest he/she gets error asking to contact administrator.
Console: **guysmiley8 attempted to take quest "null" but something went**

**If applicable, can you provide an example quest from quests.yml _using Pastebin.com_?**
https://pastebin.com/GNL6VJQx

**Comment by lyimpike** [comment]
This is on a FRESH install of the quests plugin. No other plugins are interfering.
No players accepted the quest while I was working on it either.

**Comment by PikaMug** [comment]
This could happen when the quest is...

1. currently being worked on in the editor
2. finished in the editor but forgot to run _/questadmin reload_
3. offered by an NPC, but the dies or relogs during the conversation times out

Have any other users experienced this error? Has it happened to guysmiley8 more than once?

**Comment by lyimpike** [comment]
This happens when there is a quest already set up and a fresh spawned player tries to accept the quest.

The quest was set up.
Plugin was reloaded.
This was the players first time interacting with the NPC.

This happens to all my new players until I manually /questadmin reset %p%
After that, it never happens again to the same player.

**Comment by lyimpike** [comment]
I am almost 100% certain there is some kind of permission that I am missing, NO other plugins and when fresh players they cant access the quest until I manually reset their progress, They can then complete all the stages until the final stage where they need /op to completer it.

This is on 3.8.0 and lower for 1.14.4

**Comment by SXRWahrheit** [comment]
Please provide a latest.log on pastebin from startup to error occurrence.

**Comment by PikaMug** [comment]
@SXRWahrheit Thanks, but I don't think the console error has a stacktrace this time.

@lyimpike I've got a lead on the issue, but I need to ask a few more questions. How many quests (including default ones) are currently in your quests.yml file?

**Comment by lyimpike** [comment]
Only mine, the default are untouched.

**Comment by PikaMug** [comment]
Thank you. Please try the following build once available:
https://ci.codemc.org/job/PikaMug/job/Quests/30/

If the issue persists, does setting `ask-confirmation: false` in config.yml resolve the issue?

**Comment by lyimpike** [comment]
Am I just downloading that 3.8.0? And is it available now?

I'll give it a try on a fresh server tomorrow.

**Comment by PikaMug** [comment]
Yes. You want the “Quests-3.8.0.jar” link.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by lyimpike** [comment]
Still saying it requires an admin. Going to test it on a fresh clean version

**Comment by PikaMug** [comment]
I've pushed a commit which I believe resolves this issue. Like before, please try the new build:
https://ci.codemc.org/job/PikaMug/job/Quests/31/

If the issue persists, let me know if setting `ask-confirmation: false` in config.yml has any effect.

**Comment by lyimpike** [comment]
@PikaMug works, I found the incompatability. It's LuckPerms.
Do you know why this would be causing the issue.

**Comment by lyimpike** [comment]
I am contacting Luck Perms as well just to see if they have any explanation.

**Comment by PikaMug** [comment]
That's surprising considering there are no permissions checks at that point of the conversation (where the "something went wrong" prompt appears). We also have many users that use LuckPerms without incident.

I'll try to set up a test server with LuckPerms installed. Would you be able to provide a list of permissions you've used for Quests?

**Comment by lyimpike** [comment]
Ill send you my LuckPerms permissions
![image](https://user-images.githubusercontent.com/25600834/65727175-d91a1b00-e06b-11e9-8993-02b6d3dd358d.png)
![image](https://user-images.githubusercontent.com/25600834/65727191-e33c1980-e06b-11e9-84cc-d09bac03ff6e.png)
![image](https://user-images.githubusercontent.com/25600834/65727201-eafbbe00-e06b-11e9-89cd-bf69aee35475.png)
![image](https://user-images.githubusercontent.com/25600834/65727215-f0f19f00-e06b-11e9-89f0-edb0a59fa975.png)

**Comment by lyimpike** [comment]
I'd also like to aplogise for my review, I am working to change it.

**Comment by lyimpike** [comment]
Update: Looks like it is just LuckPerm's causing the issue. bPerms works fine. @SXRWahrheit claimed he was able to get LuckPerms and Quests working together, but that was on an older version. I am really hopeing you can find a fix for this.

**Comment by PikaMug** [comment]
Not seeing any Quests permissions in your screenshots, so I assumed you didn't set any.

I created a fresh 1.14 server with only Quests 3.8.0-b32, Citizens 2.0.25-b1737 and LuckPerms 4.4.1 installed, then added a NPC and your "Adventure Begins" quest. Proceeding without _/op_, I was able to both accept and complete the quest without having set any permissions at all. I tested that LuckPerms was working by running _/quests list_ and getting blocked. Only by giving myself the "quests.list" permission was I able to use the command.

All I can figure is that this could be some system-dependent bug within LuckPerms, since switching to bPerms resolved the issue for you. Neither myself nor Wahrheit are seemingly capable of recreating the bug in our environments. At this point, it appears whatever is going on here will be up to the LuckPerms team to figure out.
